### ISSUE TITLE
call of pspm_butter & pspm_prepdata failed

### ISSUE THREAD

**Issue by kasimacar** [bachlab/PsPM#79]
Hi there,

I am following the GLM for SCR tutorial: Appraisal data, but run in to this issue when trying to import data:

"23-Dec-2019 19:48:58 - Running job #2
------------------------------------------------------------------------
23-Dec-2019 19:48:58 - Running 'Import'

Pulse data was converted to waveform with a sampling rate of 5000.00 Hz, to allow 10-fold oversampling.
Warning: call of pspm_butter failed 
> In pspm_prepdata (line 102)
  In pspm_pulse_convert (line 75)
  In pspm_get_spike (line 82)
  In pspm_import (line 199)
  In pspm_cfg_run_import (line 97)
  In cfg_run_cm (line 29)
  In cfg_util>local_runcj (line 1708)
  In cfg_util (line 968)
  In cfg_ui>MenuFileRun_Callback (line 703)
  In gui_mainfcn (line 95)
  In cfg_ui (line 53) 
Warning: call of pspm_prepdata failed 
> In pspm_pulse_convert (line 77)
  In pspm_get_spike (line 82)
  In pspm_import (line 199)
  In pspm_cfg_run_import (line 97)
  In cfg_run_cm (line 29)
  In cfg_util>local_runcj (line 1708)
  In cfg_util (line 968)
  In cfg_ui>MenuFileRun_Callback (line 703)
  In gui_mainfcn (line 95)
  In cfg_ui (line 53) 
 done.
23-Dec-2019 19:49:02 - Done    'Import'
23-Dec-2019 19:49:02 - Done"

Although this actually creates the file "pspm_trsp_1_25.mat", I seem to get errors when trying to specify the first-level GLM, which does not create "glm_25.mat":

"------------------------------------------------------------------------
23-Dec-2019 19:55:01 - Running job #4
------------------------------------------------------------------------
23-Dec-2019 19:55:01 - Running 'GLM for SCR'
Computing GLM: /Users/kash/Downloads/Tutorial_data_GLM/glm_25.mat ...
Getting data ....
Preparing & inverting model ... Warning: call of pspm_butter failed 
> In pspm_prepdata (line 102)
  In pspm_glm (line 484)
  In pspm_cfg_run_glm_scr (line 38)
  In cfg_run_cm (line 29)
  In cfg_util>local_runcj (line 1708)
  In cfg_util (line 968)
  In cfg_ui>MenuFileRun_Callback (line 703)
  In gui_mainfcn (line 95)
  In cfg_ui (line 53) 
Warning: Failed to filter data 
> In pspm_glm (line 485)
  In pspm_cfg_run_glm_scr (line 38)
  In cfg_run_cm (line 29)
  In cfg_util>local_runcj (line 1708)
  In cfg_util (line 968)
  In cfg_ui>MenuFileRun_Callback (line 703)
  In gui_mainfcn (line 95)
  In cfg_ui (line 53) 
23-Dec-2019 19:55:02 - Done    'GLM for SCR'
23-Dec-2019 19:55:02 - Done"

Any ideas why this is happening?

Attaching all the batch files in case I have made an obvious mistake.

[batch-files.zip](https://github.com/bachlab/PsPM/files/3995944/batch-files.zip)

**Comment by kasimacar** [comment]
I also tried the Non-Linear SCR tutorial, with the provided data set on sourceforge, and get this:

"------------------------------------------------------------------------
23-Dec-2019 21:32:54 - Running job #1
------------------------------------------------------------------------
23-Dec-2019 21:32:54 - Running 'Non-Linear Model'
Warning: call of pspm_butter failed 
> In pspm_prepdata (line 102)
  In pspm_dcm (line 443)
  In pspm_cfg_run_dcm (line 131)
  In cfg_run_cm (line 29)
  In cfg_util>local_runcj (line 1708)
  In cfg_util (line 968)
  In cfg_ui>MenuFileRun_Callback (line 703)
  In gui_mainfcn (line 95)
  In cfg_ui (line 53) 
23-Dec-2019 21:32:56 - Done    'Non-Linear Model'
23-Dec-2019 21:32:56 - Done"

I have attached the batch-file for that as well.

OS: macOS Catalina
Matlab: R2019b
PsPM: 4.2.1

[batch_model_s1.mat.zip](https://github.com/bachlab/PsPM/files/3996226/batch_model_s1.mat.zip)

**Comment by kasimacar** [comment]
Also, in the non-linear tutorial, it says there should be filtered SCR data named: 'tpspm_s...',
However, I only find files named 'tscr_s..." in a folder called 'scr', could this be related to the issue I'm having?

**Comment by eozd** [comment]
Hi Kasim,

Thank you for bringing this issue to our attention. We will analyze your bug report, and will get back here with more information as soon as possible. Please excuse our week long delay in writing a response due to the holiday season.

**Comment by kasimacar** [comment]
Hi Eshref,

No worries, thank you for your response. Looking forward to hear your solution to this.

Kindly,
Kasim

**Comment by irojkov-ph** [comment]
Hi Kasim,

Please excuse us again for taking so long to respond to your issue. 

Here is what we think about it:
It is probably caused by the fact that the Signal Processing Toolbox is missing on your computer, so PsPM calls pspm_butter instead of the Matlab function butter. pspm_butter stores some often-used filters and probably the filter you want is not contained. Could you please confirm us that this is the case ? If not we will make further investigations.

Best,
Ivan.

**Comment by irojkov-ph** [comment]
> Also, in the non-linear tutorial, it says there should be filtered SCR data named: 'tpspm_s...',
> However, I only find files named 'tscr_s..." in a folder called 'scr', could this be related to the issue I'm having?

For this issue, we agree with you that the data are inconsistent with the tutorial. We've changed the tutorial but forgot to modify the data in consequence.
Thank you for pointing it out. We will update the data file names and re-upload the correct data soon. In the meantime, you can use the `tscr_s...` data as the `tpspm_s...` one.

**Comment by kasimacar** [comment]
> Hi Kasim,
> 
> Please excuse us again for taking so long to respond to your issue.
> 
> Here is what we think about it:
> It is probably caused by the fact that the Signal Processing Toolbox is missing on your computer, so PsPM calls pspm_butter instead of the Matlab function butter. pspm_butter stores some often-used filters and probably the filter you want is not contained. Could you please confirm us that this is the case ? If not we will make further investigations.
> 
> Best,
> Ivan.

Thank you for your response, Ivan.

That is what I figured after initially getting that error. So I completely removed Matlab, reinstalled it again (this time with Signal Processing Toolbox), and re-ran the tutorial. But I got the same error and only after this I reported the issue here.

However, I recently made a clean macOS install on my computer, and installed Matlab with the Signal Processing Toolbox again, and that seemed to solve the problem (no idea why it should matter). Everything works as it should now.

Thank you for taking your time and addressing my issue.

Kindly,
Kasim

**Comment by irojkov-ph** [comment]
Perfect, I'm glad to here that ! :)
I close the issue then.
Do not hesitate to open a new issue if you encounter another problem, we should be more reactive now that the end of the year break is over !
